gateway-ha/src/main/java/io/trino/gateway/baseapp/AppModule.java

```diff
@@ -46,9 +47,13 @@ protected int getApplicationPort()
                         .map(SimpleServerFactory::getConnector);
 
         return connectors
-                .filter(connector -> connector.getClass().isAssignableFrom(HttpConnectorFactory.class))
-                .map(connector -> (HttpConnectorFactory) connector)
-                .mapToInt(HttpConnectorFactory::getPort)
+                .filter(connector -> connector instanceof HttpConnectorFactory)
```

Comment:
When do we get non-Http(s)ConnectorFactory here? We should throw an exception if it's unexpected status.

Reply:
So i think if someone configures any non-http(s) type I see that dropwizard throws an exception on the startup that its unable to parse the config itself. So i think we should be good here. please correct me if you meant something else.

Reply:
I mean we should throw an exception explicitly instead of doing `filter` if it's the unexpected status.